File: tests/leetcode/x/pascal/0121.pas
program Main;

function MaxProfit(prices: array of LongInt): LongInt;
var i, minPrice, best, profit: LongInt;
begin
  if Length(prices) = 0 then
  begin
    MaxProfit := 0;
    Exit;
  end;
  minPrice := prices[0];
  best := 0;
  for i := 1 to High(prices) do
  begin
    profit := prices[i] - minPrice;
    if profit > best then best := profit;
    if prices[i] < minPrice then minPrice := prices[i];
  end;
  MaxProfit := best;
end;

var t, tc, n, i: LongInt; prices: array of LongInt;
begin
  if EOF then Halt(0);
  ReadLn(t);
  for tc := 1 to t do begin
    ReadLn(n);
    SetLength(prices, n);
    for i := 0 to n - 1 do ReadLn(prices[i]);
    Write(MaxProfit(prices));
    if tc < t then WriteLn;
  end;
end.

File: tests/leetcode/x/c/0121.c
#include <stdio.h>

static int max_profit(int *prices, int n) {
    if (n == 0) return 0;
    int min_price = prices[0], best = 0;
    for (int i = 1; i < n; i++) {
        int profit = prices[i] - min_price;
        if (profit > best) best = profit;
        if (prices[i] < min_price) min_price = prices[i];
    }
    return best;
}

int main(void) {
    int t;
    if (scanf("%d", &t) != 1) return 0;
    for (int tc = 0; tc < t; tc++) {
        int n;
        int prices[100005];
        scanf("%d", &n);
        for (int i = 0; i < n; i++) scanf("%d", &prices[i]);
        printf("%d", max_profit(prices, n));
        if (tc + 1 < t) putchar('\n');
    }
    return 0;
}
